source/js/addons.js

```diff
@@ -0,0 +1,154 @@
+var apiUrl = "http://api.playnite.link/api/addons";
+
+var addons = [];
+var addonsGameLibrary = [];
+var addonsMetadataProvider = [];
+var addonsGeneric = [];
+var addonsThemeDesktop = [];
+var addonsThemeFullscreen = [];
+
+function sortByName(a, b) {
+    var aName = a.name.toLowerCase();
+    var bName = b.name.toLowerCase();
+    return ((aName < bName) ? -1 : ((aName > bName) ? 1 : 0));
+}
+
+$(document).ready(function () {
+    getAddonList();
+
+    $("#searchBox").keyup(function (data) {
+        if (data.target.value.length > 0) {
+            var searchResult = search(data.target.value);
+            resortAddons(searchResult.map(function (val) { return val.item }));
+            rerenderAddons();
+        }
+        else {
+            resortAddons(addons);
+            rerenderAddons();
+        }
+    });
+});
+
+function getAddonList() {    
+    $.getJSON(apiUrl, function (data) {
+        addons = data.data.sort(sortByName);
+        resortAddons(data.data);
+        rerenderAddons();
+    });
+}
+
+function resortAddons(data) {
+    addonsGameLibrary = [];
+    addonsMetadataProvider = [];
+    addonsGeneric = [];
+    addonsThemeDesktop = [];
+    addonsThemeFullscreen = [];
+    $.each(data, function (_key, val) {
+        switch (val.type) {
+            case 1:
+                addonsMetadataProvider.push(val);
+                break;
+            case 2:
+                addonsGeneric.push(val);
+                break;
+            case 3:
+                addonsThemeDesktop.push(val);
+                break;
+            case 4:
+                addonsThemeFullscreen.push(val);
+                break;
+            case undefined:
+                addonsGameLibrary.push(val);
+                break;
+            default:
+        }
+    });
+}
+
+function rerenderAddons() {
+    renderAddonsSection(addonsMetadataProvider, "addonsMetadataProvider", "Metadata Sources");
+    renderAddonsSection(addonsGeneric, "addonsGeneric", "Generic");
+    renderAddonsSection(addonsThemeDesktop, "addonsThemeDesktop", "Themes Desktop");
+    renderAddonsSection(addonsThemeFullscreen, "addonsThemeFullscreen", "Themes Fullscreen");
+    renderAddonsSection(addonsGameLibrary, "addonsGameLibrary", "Libraries");
+}
+
+function renderAddonsSection(data, type, name) {
+    var htmlItems = [];
+    if (data.length > 0) {
+        $.each(data, function (_key, val) {
+            htmlItems.push(" \
+            <li class='list-group-item'> \
+                <div class='row'> \
+                    <div class='col-sm-2'> \
+                        " + (val.iconUrl != undefined ? "<img src=" + val.iconUrl + " width=120></img>" : "") + "\
+                    </div> \
+                    <div class='col-sm-9'> \
+                        <h4>" + val.name + "</h4> \
+                        <p>" + (val.description == undefined ? val.shortDescription : val.description) + "</p> \
```

Comment:
Building HTML with string concatenation like this is unsafe unless you're sure the input has been sanitized and is safe to use in each context (`<` and `>` will be evaluated when you insert text into the contents of tags, and `"` is unsafe when you use the value in a double quoted attribute).

Also, it would be nice to preserve literal newlines in the descriptions. To keep them you can either replace newlines with `<br>` or style with something like `white-space: pre-line`.

Reply:
The string are not really external, as Crow controls what gets into the addon database.

I can sanitize them on frontend, but I'd expect that API already does that anyway. Does backend sanitize strings @JosefNemec?

I don't think `"` would be a problem in the way I uses outside values, but I'll double check and come back with a confirmation. 

The newline stuff is a great suggestion, thanks:).

Reply:
Backend doesn't sanitize anything. But I think I would notice people injecting JS into descriptions.

Reply:
I am writing bot to automatically check those add-on PRs so I can add check for that.

Reply:
Just to be on the safe side, I've just sanitized and/or encoded everything external.

Reply:
Ah, I didn't realize these were only extension descriptions from PlayniteAddonDatabase, not the developer provided installation manifests which are less trusted. Yes, if there's a good PR review process you won't get anything malicious through it.

@JosefNemec If you're planning a bot, one option that works well is JSON Schema (yes it works against YAML too). The schema can specify the entire YAML structure and even can use regexes to restrict allowed characters in different fields. I once set up a pipeline to enforce this with CI once on a project.

Reply:
Ah, I didn't realize these were only extension descriptions from PlayniteAddonDatabase, not the developer provided installation manifests which are less trusted. Yes, if there's a good PR review process you won't get anything malicious through it.

@JosefNemec If you're planning a bot, one option that works well is JSON Schema (yes it works against YAML too). The schema can specify the entire YAML structure and even can use regexes to restrict allowed characters in different fields. I once set up a pipeline to enforce this with CI once on a project.